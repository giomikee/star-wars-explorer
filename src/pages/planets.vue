<template>
  <v-container>
    <h2>Planets Database</h2>
  </v-container>

  <SearchNameInput
    :is-loading="arePlanetsEmpty"
    @name-searched="setSearchedName"
  />

  <SortButtons
    :is-loading="arePlanetsEmpty"
    @sorted:date="sortPlanetsByDate"
    @sorted:name="sortPlanetsByName"
  />

  <TableWrapper
    :current-page="currentPage"
    :is-loading="arePlanetsEmpty"
    :pages="itemsPages"
    @page-changed="updateCurrentPage"
  >
    <PlanetsTable :current-page="currentPage" />
  </TableWrapper>
</template>

<script setup lang="ts">
  import PlanetsTable from '../components/PlanetsTable/PlanetsTable.vue'
  import SearchNameInput from '../components/SearchNameInput/SearchNameInput.vue'
  import TableWrapper from '../components/TableWrapper/TableWrapper.vue'
  import { usePlanetsPage } from './composables'

  const {
    arePlanetsEmpty,
    currentPage,
    itemsPages,
    setSearchedName,
    sortPlanetsByDate,
    sortPlanetsByName,
    updateCurrentPage,
  } = usePlanetsPage()
</script>
