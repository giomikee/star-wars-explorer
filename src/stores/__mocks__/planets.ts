import type { Planet } from '../../interfaces'

const mockedDate = new Date('2014-12-09T13:50:49.641000Z')

export const mockedPlanets: Planet[] = [
  {
    name: 'Tatooine',
    rotation_period: '23',
    orbital_period: '304',
    diameter: '10465',
    climate: 'arid',
    gravity: '1 standard',
    terrain: 'desert',
    surface_water: '1',
    population: '200000',
    residents: [
      'https://swapi.info/api/people/1',
      'https://swapi.info/api/people/2',
      'https://swapi.info/api/people/4',
      'https://swapi.info/api/people/6',
      'https://swapi.info/api/people/7',
      'https://swapi.info/api/people/8',
      'https://swapi.info/api/people/9',
      'https://swapi.info/api/people/11',
      'https://swapi.info/api/people/43',
      'https://swapi.info/api/people/62',
    ],
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/4',
      'https://swapi.info/api/films/5',
      'https://swapi.info/api/films/6',
    ],
    created: '2014-12-09T13:50:49.641000Z',
    edited: mockedDate,
    url: 'https://swapi.info/api/planets/1',
  },
  {
    name: 'Planet2',
    rotation_period: '23',
    orbital_period: '304',
    diameter: '10465',
    climate: 'arid',
    gravity: '1 standard',
    terrain: 'desert',
    surface_water: '1',
    population: '200000',
    residents: [
      'https://swapi.info/api/people/1',
      'https://swapi.info/api/people/2',
      'https://swapi.info/api/people/4',
      'https://swapi.info/api/people/6',
      'https://swapi.info/api/people/7',
      'https://swapi.info/api/people/8',
      'https://swapi.info/api/people/9',
      'https://swapi.info/api/people/11',
      'https://swapi.info/api/people/43',
      'https://swapi.info/api/people/62',
    ],
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/4',
      'https://swapi.info/api/films/5',
      'https://swapi.info/api/films/6',
    ],
    created: '2014-12-09T13:50:49.641000Z',
    edited: mockedDate,
    url: 'https://swapi.info/api/planets/1',
  },
  {
    name: 'Planet3',
    rotation_period: '23',
    orbital_period: '304',
    diameter: '10465',
    climate: 'arid',
    gravity: '1 standard',
    terrain: 'desert',
    surface_water: '1',
    population: '200000',
    residents: [
      'https://swapi.info/api/people/1',
      'https://swapi.info/api/people/2',
      'https://swapi.info/api/people/4',
      'https://swapi.info/api/people/6',
      'https://swapi.info/api/people/7',
      'https://swapi.info/api/people/8',
      'https://swapi.info/api/people/9',
      'https://swapi.info/api/people/11',
      'https://swapi.info/api/people/43',
      'https://swapi.info/api/people/62',
    ],
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/4',
      'https://swapi.info/api/films/5',
      'https://swapi.info/api/films/6',
    ],
    created: '2014-12-09T13:50:49.641000Z',
    edited: mockedDate,
    url: 'https://swapi.info/api/planets/1',
  },
  {
    name: 'Planet4',
    rotation_period: '23',
    orbital_period: '304',
    diameter: '10465',
    climate: 'arid',
    gravity: '1 standard',
    terrain: 'desert',
    surface_water: '1',
    population: '200000',
    residents: [
      'https://swapi.info/api/people/1',
      'https://swapi.info/api/people/2',
      'https://swapi.info/api/people/4',
      'https://swapi.info/api/people/6',
      'https://swapi.info/api/people/7',
      'https://swapi.info/api/people/8',
      'https://swapi.info/api/people/9',
      'https://swapi.info/api/people/11',
      'https://swapi.info/api/people/43',
      'https://swapi.info/api/people/62',
    ],
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/4',
      'https://swapi.info/api/films/5',
      'https://swapi.info/api/films/6',
    ],
    created: '2014-12-09T13:50:49.641000Z',
    edited: mockedDate,
    url: 'https://swapi.info/api/planets/1',
  },
  {
    name: 'Planet5',
    rotation_period: '23',
    orbital_period: '304',
    diameter: '10465',
    climate: 'arid',
    gravity: '1 standard',
    terrain: 'desert',
    surface_water: '1',
    population: '200000',
    residents: [
      'https://swapi.info/api/people/1',
      'https://swapi.info/api/people/2',
      'https://swapi.info/api/people/4',
      'https://swapi.info/api/people/6',
      'https://swapi.info/api/people/7',
      'https://swapi.info/api/people/8',
      'https://swapi.info/api/people/9',
      'https://swapi.info/api/people/11',
      'https://swapi.info/api/people/43',
      'https://swapi.info/api/people/62',
    ],
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/4',
      'https://swapi.info/api/films/5',
      'https://swapi.info/api/films/6',
    ],
    created: '2014-12-09T13:50:49.641000Z',
    edited: mockedDate,
    url: 'https://swapi.info/api/planets/1',
  },
  {
    name: 'Planet6',
    rotation_period: '23',
    orbital_period: '304',
    diameter: '10465',
    climate: 'arid',
    gravity: '1 standard',
    terrain: 'desert',
    surface_water: '1',
    population: '200000',
    residents: [
      'https://swapi.info/api/people/1',
      'https://swapi.info/api/people/2',
      'https://swapi.info/api/people/4',
      'https://swapi.info/api/people/6',
      'https://swapi.info/api/people/7',
      'https://swapi.info/api/people/8',
      'https://swapi.info/api/people/9',
      'https://swapi.info/api/people/11',
      'https://swapi.info/api/people/43',
      'https://swapi.info/api/people/62',
    ],
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/4',
      'https://swapi.info/api/films/5',
      'https://swapi.info/api/films/6',
    ],
    created: '2014-12-09T13:50:49.641000Z',
    edited: mockedDate,
    url: 'https://swapi.info/api/planets/1',
  },
  {
    name: 'Planet7',
    rotation_period: '23',
    orbital_period: '304',
    diameter: '10465',
    climate: 'arid',
    gravity: '1 standard',
    terrain: 'desert',
    surface_water: '1',
    population: '200000',
    residents: [
      'https://swapi.info/api/people/1',
      'https://swapi.info/api/people/2',
      'https://swapi.info/api/people/4',
      'https://swapi.info/api/people/6',
      'https://swapi.info/api/people/7',
      'https://swapi.info/api/people/8',
      'https://swapi.info/api/people/9',
      'https://swapi.info/api/people/11',
      'https://swapi.info/api/people/43',
      'https://swapi.info/api/people/62',
    ],
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/4',
      'https://swapi.info/api/films/5',
      'https://swapi.info/api/films/6',
    ],
    created: '2014-12-09T13:50:49.641000Z',
    edited: mockedDate,
    url: 'https://swapi.info/api/planets/1',
  },
  {
    name: 'Planet8',
    rotation_period: '23',
    orbital_period: '304',
    diameter: '10465',
    climate: 'arid',
    gravity: '1 standard',
    terrain: 'desert',
    surface_water: '1',
    population: '200000',
    residents: [
      'https://swapi.info/api/people/1',
      'https://swapi.info/api/people/2',
      'https://swapi.info/api/people/4',
      'https://swapi.info/api/people/6',
      'https://swapi.info/api/people/7',
      'https://swapi.info/api/people/8',
      'https://swapi.info/api/people/9',
      'https://swapi.info/api/people/11',
      'https://swapi.info/api/people/43',
      'https://swapi.info/api/people/62',
    ],
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/4',
      'https://swapi.info/api/films/5',
      'https://swapi.info/api/films/6',
    ],
    created: '2014-12-09T13:50:49.641000Z',
    edited: mockedDate,
    url: 'https://swapi.info/api/planets/1',
  },
  {
    name: 'Planet9',
    rotation_period: '23',
    orbital_period: '304',
    diameter: '10465',
    climate: 'arid',
    gravity: '1 standard',
    terrain: 'desert',
    surface_water: '1',
    population: '200000',
    residents: [
      'https://swapi.info/api/people/1',
      'https://swapi.info/api/people/2',
      'https://swapi.info/api/people/4',
      'https://swapi.info/api/people/6',
      'https://swapi.info/api/people/7',
      'https://swapi.info/api/people/8',
      'https://swapi.info/api/people/9',
      'https://swapi.info/api/people/11',
      'https://swapi.info/api/people/43',
      'https://swapi.info/api/people/62',
    ],
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/4',
      'https://swapi.info/api/films/5',
      'https://swapi.info/api/films/6',
    ],
    created: '2014-12-09T13:50:49.641000Z',
    edited: mockedDate,
    url: 'https://swapi.info/api/planets/1',
  },
  {
    name: 'Planet10',
    rotation_period: '23',
    orbital_period: '304',
    diameter: '10465',
    climate: 'arid',
    gravity: '1 standard',
    terrain: 'desert',
    surface_water: '1',
    population: '200000',
    residents: [
      'https://swapi.info/api/people/1',
      'https://swapi.info/api/people/2',
      'https://swapi.info/api/people/4',
      'https://swapi.info/api/people/6',
      'https://swapi.info/api/people/7',
      'https://swapi.info/api/people/8',
      'https://swapi.info/api/people/9',
      'https://swapi.info/api/people/11',
      'https://swapi.info/api/people/43',
      'https://swapi.info/api/people/62',
    ],
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/4',
      'https://swapi.info/api/films/5',
      'https://swapi.info/api/films/6',
    ],
    created: '2014-12-09T13:50:49.641000Z',
    edited: mockedDate,
    url: 'https://swapi.info/api/planets/1',
  },
  {
    name: 'Planet11',
    rotation_period: '23',
    orbital_period: '304',
    diameter: '10465',
    climate: 'arid',
    gravity: '1 standard',
    terrain: 'desert',
    surface_water: '1',
    population: '200000',
    residents: [
      'https://swapi.info/api/people/1',
      'https://swapi.info/api/people/2',
      'https://swapi.info/api/people/4',
      'https://swapi.info/api/people/6',
      'https://swapi.info/api/people/7',
      'https://swapi.info/api/people/8',
      'https://swapi.info/api/people/9',
      'https://swapi.info/api/people/11',
      'https://swapi.info/api/people/43',
      'https://swapi.info/api/people/62',
    ],
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/4',
      'https://swapi.info/api/films/5',
      'https://swapi.info/api/films/6',
    ],
    created: '2014-12-09T13:50:49.641000Z',
    edited: mockedDate,
    url: 'https://swapi.info/api/planets/1',
  },
  {
    name: 'Planet12',
    rotation_period: '23',
    orbital_period: '304',
    diameter: '10465',
    climate: 'arid',
    gravity: '1 standard',
    terrain: 'desert',
    surface_water: '1',
    population: '200000',
    residents: [
      'https://swapi.info/api/people/1',
      'https://swapi.info/api/people/2',
      'https://swapi.info/api/people/4',
      'https://swapi.info/api/people/6',
      'https://swapi.info/api/people/7',
      'https://swapi.info/api/people/8',
      'https://swapi.info/api/people/9',
      'https://swapi.info/api/people/11',
      'https://swapi.info/api/people/43',
      'https://swapi.info/api/people/62',
    ],
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/4',
      'https://swapi.info/api/films/5',
      'https://swapi.info/api/films/6',
    ],
    created: '2014-12-09T13:50:49.641000Z',
    edited: mockedDate,
    url: 'https://swapi.info/api/planets/1',
  },
]
