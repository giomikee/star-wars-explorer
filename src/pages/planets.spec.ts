import { shallowMount } from '@vue/test-utils'
import { describe, expect, it, vi } from 'vitest'
import { usePlanetsStore } from '../stores/planets.store'
import Planets from './planets.vue'

const createWrapper = () => shallowMount(Planets)

describe('Given planets page component', () => {
  describe('When component is rendered', () => {
    it('Then it should match the snapshot', () => {
      const store = usePlanetsStore()

      vi.spyOn(store, 'getPlanetsToDisplay').mockReturnValueOnce([])

      const { element } = createWrapper()

      expect(element).toMatchSnapshot()
    })
  })
})
