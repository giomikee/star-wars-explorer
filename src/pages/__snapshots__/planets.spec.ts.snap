// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Given planets page component > When component is rendered > Then it should match the snapshot 1`] = `
<div
  data-v-app=""
>
  
  <div
    class="v-container v-locale--is-ltr"
  >
    <h2>
      Planets Database
    </h2>
  </div>
  <search-name-input-stub
    isloading="true"
  />
  <sort-buttons-stub
    isloading="true"
  />
  <table-wrapper-stub
    currentpage="1"
    isloading="true"
    pages="0"
  />
  
</div>
`;
